teps/0033-tekton-feature-gates.md

```diff
@@ -145,6 +161,581 @@ alpha features. This is not a requirement for Tekton, because even in case of
 a managed Tekton, users can be prevented access to the `tekton-pipelines`
 namespaces, where the feature flag config map is hosted.
 
+## Proposal
+
+We will tackle separately features that are controlled via new fields in the API
+and features that are only behavioral. (The alternative
+[one feature flag per new alpha field](#one-feature-flag-per-new-alpha-field) describes
+an alternative where we use the same approach for both.)
+
+In both cases we want to control access with
+[a feature flag](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#customizing-the-pipelines-controller-behavior).
+
+* For new API fields, will will gate access to all of them behind one feature flag called
+  `enable-api-fields` ([Design details: new api fields](#new-api-fields).)
+* For changes in behavior which are not configurable via API fields in our CRDs,
+  each feature will be gated by its own unique feature flag.
+  ([Design details: changes in behavior](#changes-in-behavior).)
+
+This proposal would impact all projects in [the tektoncd org](http://github.com/tektoncd) that are CRD based - if they
+find themselves in the position of wanting to add fields to existing CRDs at a lower stability level than the CRD
+itself. Today this would include:
+
+* [Tekton Pipelines](https://github.com/tektoncd/pipeline)
+* [Tekton Triggers](https://github.com/tektoncd/triggers)
+
+### Risks and Mitigations
+
+See the pros and cons sections in the [design details](#design-details)
+below.
+
+## Design Details
+
+### New API fields
+
+We gate access to all new API fields with a new
+[feature flag](https://github.com/tektoncd/pipeline/blob/master/docs/install.md#customizing-the-pipelines-controller-behavior)
+called `enable-api-fields` with possible values of:
+
+* `stable` (default) - This value indicates that only fields of the highest stability level are enabled; at the moment
+  the highest version in Tekton Pipelines is `beta`, so this would mean only beta stability fields are enabled, i.e.
+  `alpha` fields are not enabled. Once we have a `v1` version, this would mean only `v1` fields, i.e. `beta` and `alpha`
+  fields would not be enabled.
+* `beta` (not needed until `v1`) - This value indicates that only fields which are of `beta` (or greater) stability are
+  enabled, i.e. `alpha` fields are not enabled. Since `beta` is currently the highest level of stability in Tekton
+  Pipelines, this value is not needed and does not need to be added until we support (at least one) `v1` version CRD.
+* `alpha` - This value indicates that fields of all stability levels are enabled, specifically `alpha`, `beta` and
+  (once its available) `v1`.
```

Comment:
Sorry for one more table, just making sure I understand this right 🙏 

| Feature Versions -> | v1 | beta | alpha |
| --- | --- | --- | --- |
| stable | :heavy_check_mark: | | |
| beta | :heavy_check_mark: |  :heavy_check_mark: | |
| alpha | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |

* Operator sets the `enable-api-fields` to `beta` to disable all `alpha` features but allow `beta` features.
* Operator sets the `enable-api-fields` to `stable` to disable all `alpha` and `beta` features.

Reply:
no apology needed! that looks right

technically before we have v1 it'll be more like:

| Feature Versions ->  | beta | alpha |
| ---  | --- | --- |
| stable | :heavy_check_mark: | |
| alpha | :heavy_check_mark: | :heavy_check_mark: |

and then your table will be the behavior after the v1 release - im going to add these tables to the TEP to make it more clear